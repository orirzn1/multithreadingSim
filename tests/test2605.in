L12
S5
O11
N7

T0
I@0x00000000
SUBI $3, $5, 0x2
STORE $6, $3, $3
ADD $2, $7, $3
ADD $6, $3, $4
HALT $0

T1
I@0x00000000
ADDI $4, $4, 0x1
ADDI $1, $7, 0x1
STORE $3, $0, $6
ADDI $1, $7, 0x2
SUB $1, $3, $6
STORE $6, $3, $4
LOAD $5, $0, $2
HALT $0

T2
I@0x00000000
STORE $6, $4, $4
ADDI $7, $4, 0x4
SUBI $4, $6, 0x4
SUBI $0, $0, 0x1
SUBI $0, $4, 0x1
ADD $2, $5, $2
SUBI $7, $1, 0x1
SUBI $2, $7, 0x3
STORE $7, $2, 0x2
SUB $0, $0, $6
HALT $0

T3
I@0x00000000
ADDI $3, $6, 0x2
SUB $2, $2, $7
STORE $4, $7, $4
ADDI $6, $3, 0x3
LOAD $2, $4, $2
ADDI $3, $5, 0x4
ADD $2, $7, $2
LOAD $0, $1, $4
STORE $4, $1, $3
HALT $0

T4
I@0x00000000
ADDI $4, $6, 0x2
SUB $1, $5, $6
ADDI $1, $2, 0x4
ADD $7, $2, $6
ADD $4, $6, $0
SUB $0, $6, $7
SUB $5, $2, $0
ADDI $5, $4, 0x3
LOAD $7, $0, $6
ADDI $0, $1, 0x2
HALT $0

T5
I@0x00000000
ADD $7, $0, $6
ADD $6, $5, $3
SUB $7, $4, $6
STORE $3, $6, $6
ADDI $1, $6, 0x4
SUB $5, $5, $3
HALT $0

T6
I@0x00000000
SUB $6, $1, $2
ADDI $7, $1, 0x2
LOAD $0, $3, $2
SUBI $6, $6, 0x3
STORE $2, $4, $2
ADDI $2, $3, 0x1
LOAD $3, $2, $0
ADDI $4, $6, 0x4
SUB $4, $7, $5
STORE $6, $6, $0
HALT $0

D@0x0
0x8
0x5
0x4
0x4
0x3
