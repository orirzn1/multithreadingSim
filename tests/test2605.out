
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0xFFFFFFFB	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x4	R4 = 0x0	R5 = 0x0	R6 = 0x5	R7 = 0x0
Register file thread id 4:
	R0 = 0x6	R1 = 0x4	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x4	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x2
Blocked MT CPI for this program 4.841270

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0xFFFFFFFB	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 3:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x0	R3 = 0x4	R4 = 0x0	R5 = 0x0	R6 = 0x5	R7 = 0x0
Register file thread id 4:
	R0 = 0x6	R1 = 0x4	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0xFFFFFFFC
Register file thread id 5:
	R0 = 0x0	R1 = 0x4	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0xFFFFFFFC	R4 = 0x2	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x2
Finegrained Multithreading CPI for this program 1.174603

